### PR TITLE
Adds SNS_IK() class for using URDF & limits

### PR DESCRIPTION
**Pull request by rethink-imcmahon**
This changes adds the SNS_IK class which will construct a
KDL chain from the specified robot_description parameter.

Position and velocity limits can be read from the URDF, and overridden
by including a standard robot_description_planning/joint_limits yaml
file of lower position, upper position, max velocity and max acceleration
limits.

Finally, a new testing executable has been included which is based largely
off of trac_ik_examples tests. This will compare sns_ik against KDL and
trac_ik position solvers.

### CODE REVIEW

File: sns_ik_examples/config/joint_limits.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+joint_limits:
```
Comment (rethink-forrest) [inline sns_ik_examples/config/joint_limits.yaml]: This file is specific to baxter. Should the name reflect that?

File: sns_ik_examples/launch/test_ik_solvers.launch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <arg name="chain_end" default="right_hand" />
+  <arg name="timeout" default="0.005" />
+  <arg name="urdf_param" default="/robot_description" />
+  <arg name="load_robot_description" default="true"/>
+  <param if="$(arg load_robot_description)" name="$(arg urdf_param)"
+    command="cat $(find baxter_description)/urdf/baxter.urdf"/>
+  <rosparam command="load" ns="$(arg urdf_param)_planning"
+    file="$(find sns_ik_examples)/config/joint_limits.yaml" />
+  <!-- param if="$(arg load_robot_description)" name="$(arg urdf_param)"
+    command="cat $(find sawyer_pv_description)/xml/sawyer_pv_urdf.xml"/ -->
```
Comment (rethink-forrest) [inline sns_ik_examples/launch/test_ik_solvers.launch]: Remove the reference to sawyer files.

File: sns_ik_examples/src/ik_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+{
+  double f = (double)rand() / RAND_MAX;
+  return min + f * (max - min);
+}
+
+
+void test(ros::NodeHandle& nh, double num_samples, std::string chain_start, std::string chain_end, double timeout, std::string urdf_param)
+{
+
+  double eps = 1e-5;
```
Comment (rethink-forrest) [inline sns_ik_examples/src/ik_tests.cpp]: This should probably be a configuration parameter, but let's not worry about that for now.

File: sns_ik_examples/src/test_sns_ik.cpp [hunk 4]
```diff
@@ -35,9 +36,9 @@ int main(int argc, char** argv) {
 
   VectorD l = VectorD::Ones(7);
 
-  SNSVelocityIK ikVelSolver(7, 0.01);
-  ikVelSolver.setJointsCapabilities(-3.0*l, 3.0*l, l, 0.5*l);
-  ikVelSolver.getJointVelocity(&jointVelocity, sot, joints);
+  std::shared_ptr<SNSVelocityIK> ikVelSolver(new SNSVelocityIK(7, 0.01));
```
Comment (rethink-forrest) [inline sns_ik_examples/src/test_sns_ik.cpp]: Does this need to be a pointer here?

File: sns_ik_lib/src/sns_position_ik.cpp [hunk 5]
```diff
@@ -21,9 +21,9 @@
 
 #include <iostream>
 
-using namespace sns_ik;
+namespace sns_ik{
```
Comment (rethink-forrest) [inline sns_ik_lib/src/sns_position_ik.cpp]: Preference against using namespace?

File: sns_ik_examples/config/joint_limits.yaml (hunk 1 above)
Comment (rethink-imcmahon) [inline sns_ik_examples/config/joint_limits.yaml]: Sure, that's fine. It's just an example.

File: sns_ik_examples/launch/test_ik_solvers.launch (hunk 2 above)
Comment (rethink-imcmahon) [inline sns_ik_examples/launch/test_ik_solvers.launch]: noted.

File: sns_ik_examples/src/test_sns_ik.cpp (hunk 4 above)
Comment (rethink-imcmahon) [inline sns_ik_examples/src/test_sns_ik.cpp]: yup! it;'s used in the position solver below, so this should be a pointer

File: sns_ik_lib/src/sns_position_ik.cpp (hunk 5 above)
Comment (rethink-imcmahon) [inline sns_ik_lib/src/sns_position_ik.cpp]: Yeah, but not a very strong one. I believe this method is slightly preferred to be more explicit about showing where the namespace applies for a library.

**Comment by rethink-imcmahon** [comment]
Updated based on review comments.

**Comment by rethink-forrest** [comment]
For it to build on my computer, you add following dependence in sns_ik_examples/package.xml

<build_depend>trac_ik_lib</build_depend>

**Pull request merged**